method_comparison/peft_bench/README.md

```diff
@@ -0,0 +1,216 @@
+
+## Quick Start
+
+### Running a Single Experiment
+
+```bash
+# From the peft_bench directory
+python run.py experiments/lora/lora_r8 --verbose
+```
+
+### Using the Python API
+
+```python
+from method_comparison.peft_bench import run_benchmark_from_path
+
+# Run benchmark programmatically
+exit_code = run_benchmark_from_path("experiments/lora/lora_r8", verbose=True)
+```
+
+## Configuration Structure
+
+The benchmarking suite uses a hierarchical configuration system:
+
+1. **Default benchmark parameters** (`default_benchmark_params.json`) - Base configuration shared by all experiments
+2. **Experiment-specific overrides** (`benchmark_params.json` in each experiment) - Optional overrides for specific experiments  
+3. **Adapter configuration** (`adapter_config.json` in each experiment) - PEFT method parameters
+
+This structure ensures consistent comparison while allowing flexibility where needed.
+
+### Default Configuration (`default_benchmark_params.json`)
+
+Contains shared benchmark settings that apply to all experiments. Here are the key configuration fields:
+
+- `model_id`: The Hugging Face model ID to use as the base model (e.g., "facebook/opt-350m")
+- `peft_method`: The PEFT method to apply (e.g., "lora", "prefix_tuning", etc.)
+- `dtype`: Model precision ("float16", "float32", or "bfloat16")
+- `seed`: Random seed for reproducibility
+- `max_new_tokens`: Maximum number of tokens to generate during inference
+- `num_inference_runs`: Number of inference runs per prompt for statistical reliability
+- `train_batch_size`: Batch size for training
+- `train_steps`: Number of training steps
+- `num_prompt_samples`: Number of prompt samples to use per category (short/medium/long)
+
+Each experiment can override these settings by providing its own `benchmark_params.json` file.
+
+### Experiment Structure
+
+Each experiment directory should contain:
+
+1. `adapter_config.json`: PEFT adapter configuration. For details on available parameters and their meanings, refer to the [PEFT documentation](https://huggingface.co/docs/peft/main/en/developer_guides/adapters).
+
+2. (Optional) `benchmark_params.json`: Override specific benchmark parameters for this experiment.
+
+Example directory structure:
+```
+experiments/
+└── lora/
+    ├── lora_r8/                # LoRA rank 8 experiment
+    │   ├── adapter_config.json # PEFT adapter configuration
+    │   └── benchmark_params.json # Optional benchmark overrides
+    └── lora_r16/               # LoRA rank 16 experiment
+        └── adapter_config.json
+```
+
+### Experiment-Specific Overrides Example
+
+If an experiment needs different benchmark settings, create `benchmark_params.json`:
+```json
+{
+    "_comment": "Override settings for this specific experiment",
+    "max_new_tokens": 50,
+    "num_inference_runs": 15,
+    "num_prompt_samples": 2
+}
+```
+
+These parameters will override the defaults from `default_benchmark_params.json`.
```

Comment:
Let's add something like: "However, the defaults should generally not be changed to keep the results from the individual experiments comparable."